### ISSUE TITLE
Branching mistake leading to unused variable: to be detected

### ISSUE THREAD

**Issue by mhooreman** [PyCQA/pylint#3297]
Hello,

The following code
1. defines a filter which depends of an `if` condition
2. should apply the filter on a data set
3. ... but only applies the filter on one branch of the `if`, instead of the common branch:

```python
#!/usr/bin/env python
"""lorem ipsum"""

import numpy as np


def get(greater):
    """lorem ipsum"""
    ret = np.arange(10)
    if greater:
        f = ret > 5
        ret = ret[f].copy()  # The bug is here ...
    else:
        f = ret <= 5
    # ... the line of the bug shoud be here instead
    return ret


if __name__ == "__main__":
    print(get(True))
    print(get(False))
```

It would be great to have a warning issued in such case. I think that `unused-variable (W0612)` would be accurate.

Thanks a lot for your help and support.

**Comment by PCManticore** [comment]
Thanks for the report! Indeed we should be able to catch this unused variable.

**Comment by mhooreman** [comment]
Thanks a lot @PCManticore

**Comment by mhooreman** [comment]
Hello,

I'm sorting out my pending issues. Any update on this?
